### PR TITLE
minor: adding end-slashes when needed

### PR DESCRIPTION
**Pull request by Chrissdroid**
#### What kind of changes does this PR include?

- [x] Minor content fixes (broken links, typos, etc.)

#### Description

- What does this PR change? change *"small"* parts of the `/en/` docs links to include backslashes when needed to avoid unnessessary redirections.

### CODE REVIEW

File: src/pages/en/core-concepts/project-structure.md [hunk 1]
```diff
@@ -97,10 +97,10 @@ You can place CSS and JavaScript in your `public/` directory, but be aware that
...
 
-For help creating a new `package.json` file for your project, check out the [manual setup](/en/install/manual) instructions.
+For help creating a new `package.json` file for your project, check out the [manual setup](/en/install/manual/) instructions.
 
 ### `astro.config.mjs`
 
 This file is generated in every starter template and includes configuration options for your Astro project. Here you can specify integrations to use, build options, server options, and more.
 
-See the [Configuration Reference](https://docs.astro.build/en/reference/configuration-reference/#article) for details on setting configurations.
+See the [Configuration Reference](/en/reference/configuration-reference/#article) for details on setting configurations.
```
Comment (Chrissdroid) [inline src/pages/en/core-concepts/project-structure.md:106]: I also removed this static link that i forgot for some reason

File: src/pages/en/guides/publish-to-npm.md [hunk 2]
```diff
@@ -279,4 +279,4 @@ In addition to the required `astro-component` keyword, special keywords are also
 
 ## Share
 
-We encourage you to share your work, and we really do love seeing what our talented Astronauts create. Come and share what you create with us in our [Discord](https://discord.gg/YQRVveAgED) or mention [@astrodotbuild](https://twitter.com/astrodotbuild) in a Tweet!
+We encourage you to share your work, and we really do love seeing what our talented Astronauts create. Come and share what you create with us in our [Discord](https://astro.build/chat/) or mention [@astrodotbuild](https://twitter.com/astrodotbuild) in a Tweet!
```
Comment (Chrissdroid) [inline src/pages/en/guides/publish-to-npm.md:282]: And changed this discord invitation to the astro.build link

File: src/pages/en/editor-setup.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ Customize your code editor to improve the Astro developer experience and unlock
 
 ## VS Code
 
-[VS Code](https://code.visualstudio.com) is a popular code editor for web developers, built by Microsoft. The VS Code engine also powers popular in-browser code editors like [GitHub Codespaces](https://github.com/features/codespaces) and [Gitpod](https://gitpod.io).
+[VS Code](https://code.visualstudio.com/) is a popular code editor for web developers, built by Microsoft. The VS Code engine also powers popular in-browser code editors like [GitHub Codespaces](https://github.com/features/codespaces) and [Gitpod](https://gitpod.io/).
```
Comment (sarah11918) [inline src/pages/en/editor-setup.md:14]: Just checking that the codespaces link doesn't also need a trailing slash?

File: src/pages/en/guides/data-fetching.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ i18nReady: true
 
 ## `fetch()` in Astro
 
-All [Astro components](/en/core-concepts/astro-components) have access to the [global `fetch()` function](https://developer.mozilla.org/en-US/docs/Web/API/fetch) in their component script to make HTTP requests to APIs. This fetch call will be executed at build time, and the data will be available to the component template for generating dynamic HTML. 
+All [Astro components](/en/core-concepts/astro-components) have access to the [global `fetch()` function](https://developer.mozilla.org/en-US/docs/Web/API/fetch) in their component script to make HTTP requests to APIs. This fetch call will be executed at build time, and the data will be available to the component template for generating dynamic HTML.
```
Comment (sarah11918) [inline src/pages/en/guides/data-fetching.md:12]:
```suggestion
All [Astro components](/en/core-concepts/astro-components/) have access to the [global `fetch()` function](https://developer.mozilla.org/en-US/docs/Web/API/fetch) in their component script to make HTTP requests to APIs. This fetch call will be executed at build time, and the data will be available to the component template for generating dynamic HTML.
```
Added slash to Astro comopnents. What about the Mozilla fetch link?

**Comment by sarah11918** [comment]
Fabulous work, @Chrissdroid :star_struck: 

I only found one of our links (showing up in these file diffs) that I think was missing a slash, and there were a couple of external links where I wasn't sure whether they maybe needed a slash too, just by looking at them, so I asked you.  :)

File: src/pages/en/guides/publish-to-npm.md (hunk 2 above)
Comment (Chrissdroid) [reply src/pages/en/guides/publish-to-npm.md:282]: I was wandering why it was a custom link! it's fine if it goes to the docs chat, it's even better! (could i take the liberty to footnote this link so we know that i point directly to the docs channel ? (and so it's only visible on the soucecode))

File: src/pages/en/editor-setup.md (hunk 3 above)
Comment (Chrissdroid) [reply src/pages/en/editor-setup.md:14]: When i go to the link it doesn't redirect me to `https://github.com/features/codespaces/` so it's doesn't need it

File: src/pages/en/editor-setup.md [hunk 5]
```diff
@@ -34,6 +34,6 @@ Our amazing community maintains several extensions for other popular editors, in
...
 
-- [StackBlitz](https://stackblitz.com) and [CodeSandbox](https://codesandbox.io) - online editors that run in your browser, with built-in syntax highlighting support for `.astro` files. No installation or configuration required!
-- [GitHub.dev](https://github.dev) - allows you to install the Astro VS Code extension as a [web extension](https://code.visualstudio.com/api/extension-guides/web-extensions), which gives you access to only some of the full extension features. Currently, only syntax highlighting is supported.
-- [Gitpod](https://gitpod.io) - a full dev environment in the cloud that can install the official Astro VS Code Extension from Open VSX.
+- [StackBlitz](https://stackblitz.com/) and [CodeSandbox](https://codesandbox.io/) - online editors that run in your browser, with built-in syntax highlighting support for `.astro` files. No installation or configuration required!
+- [GitHub.dev](https://github.dev/) - allows you to install the Astro VS Code extension as a [web extension](https://code.visualstudio.com/api/extension-guides/web-extensions), which gives you access to only some of the full extension features. Currently, only syntax highlighting is supported.
```
[the remark this answers is not in the document]
Comment (Chrissdroid) [reply src/pages/en/editor-setup.md:38]: same here, when going to the link it doesn't redirect so it's fine

File: src/pages/en/guides/data-fetching.md (hunk 4 above)
Comment (Chrissdroid) [reply src/pages/en/guides/data-fetching.md:12]: All links to the MDN docs doesn't need trailing slashes 👍

**Comment by sarah11918** [comment]
Saw all your responses, so once that one link of ours is fixed, LGTM! Thank you! :rocket:

**Comment by Chrissdroid** [comment]
wait before merging so i can undo the discord link change

**Comment by sarah11918** [comment]
Oh, that Discord link is fine to change! Let's make all the links in the public docs website just https://astro.build/chat/

I only meant that there's one that goes right to docs for contributors (somewhere in README, I think) and I would like to keep THAT one the direct link to docs, maybe. But everything that will show up on the website should just be the regular /chat/ link, I think!

Go, go! Change it back! :smile:

**Comment by Chrissdroid** [comment]
I was wandering if it's not better to redirect user to the docs chat than the global chat but for consistency for now we could just let everything in the docs be the `https://astro.build/chat/` yeah

**Comment by Chrissdroid** [comment]
You can merge then! 👍

**Comment by sarah11918** [comment]
Yes, a user reading the site doesn't necessarily want to jump into the docs channel and talk about docs. :)  It's only a convenience for helping new CONTRIBUTORS find us, so that's why I think we're good having the general /chat/ address all over the website itself.

This is Great! Let's do it!

**Pull request merged**